File: Toal/piano_keys.py
KEY_NAMES = ['A', 'A#', 'B', 'C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#']
STEPS = len(KEY_NAMES)
NUMBER_OF_KEYS = 88;
INITIAL_FREQUENCY = 27.5;

for i in range(NUMBER_OF_KEYS):
    key_name = KEY_NAMES[i % STEPS]
    frequency = INITIAL_FREQUENCY * (2.0 ** (i / STEPS))
    print(f'{key_name}\t{frequency:10.4f}')

File: Toal/piano_keys.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main() {
  char *KEY_NAMES[] = {"A", "A#", "B", "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#"};
  int STEPS = sizeof( KEY_NAMES ) / sizeof( KEY_NAMES[0] );
  int INITIAL_FREQUENCY = 27.5;
  int NUMBER_OF_KEYS = 88;

  for (int i = 0; i < NUMBER_OF_KEYS; i++) {
    char *key_name = KEY_NAMES[i % STEPS];
    float frequency = INITIAL_FREQUENCY * ( pow( 2.0, (float)i / (float)STEPS ) ) ;
    //printf("i:  %d\n", i);
    //printf("STEPS:  %d\n", STEPS);
    //printf("i/STEPS:  %10.4f\n", ((float)i / (float)STEPS));
    printf("%s    %10.4f\n", key_name, frequency);
  }
  return 0;
}
